### PR TITLE
[openweathermap] Added support for Air Pollution API

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added support for Air Pollution API
- Added Channels `precip-probability`, `sunrise` and `sunset` to legacy daily forecast

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapAPIHandler.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public void initialize() {
         config = getConfigAs(OpenWeatherMapAPIConfiguration.class);
 
         boolean configValid = true;
-        if (config.apikey == null || config.apikey.trim().isEmpty()) {
+        if (config.apikey == null || config.apikey.isBlank()) {
```
Comment (wborn) [inline bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapAPIHandler.java:77]: The API key is a required configuration parameter so it cannot be null.

```suggestion
        if (config.apikey.isBlank()) {
```

**Review by wborn — COMMENTED** [review]
It's a nice addition to an already nice binding. :slightly_smiling_face:
The code looks great too and I only have a few comments:

File: bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/connection/OpenWeatherMapConnection.java [hunk 2]
```diff
@@ -88,6 +89,9 @@
     // UV Index (see https://openweathermap.org/api/uvi)
     private static final String UVINDEX_URL = "https://api.openweathermap.org/data/2.5/uvi";
     private static final String UVINDEX_FORECAST_URL = "https://api.openweathermap.org/data/2.5/uvi/forecast";
+    // Air Pollution (see https://openweathermap.org/api/air-pollution)
+    private static final String AIR_POLLUTION_URL = "http://api.openweathermap.org/data/2.5/air_pollution";
+    private static final String AIR_POLLUTION_FORECAST_URL = "http://api.openweathermap.org/data/2.5/air_pollution/forecast";
```
[the remark this answers is not in the document]
Comment (cweitkamp) [reply bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/connection/OpenWeatherMapConnection.java:94]: Yes, secure protocol is possible. Thanks for spotting.

**Comment by wborn** [comment]
Will you also push the fixes the resolve my comments @cweitkamp?

**Comment by cweitkamp** [comment]
Ooops, I thought I pushed it. But because of a previous rebase it did not work.

**Review by wborn — APPROVED** [review]
Thank you!

**Pull request merged**